src/main/java/io/gravitee/resource/oauth2/keycloak/OAuth2KeycloakResource.java

```diff
@@ -179,7 +179,7 @@ public void introspect(String accessToken, Handler<OAuth2Response> responseHandl
                 String body = buffer.toString();
                 if (response.statusCode() == HttpStatusCode.OK_200) {
                     JsonNode introspectPayload = readPayload(body);
-                    boolean active = introspectPayload.path("active").asBoolean(false);
+                    boolean active = introspectPayload != null && introspectPayload.path("active").asBoolean(false);
```

Comment:
Can you indicate me in which case the payload may be null?

Reply:
for example, when there'll be an IOException in readPayload there'll be null response. In realworld - if oauth-provider host will be broken or compomised or not a valid JSON will be returned for whatever reason.
We fixed your unit tests and will commit them in a while.